cpp/include/raft/sparse/linalg/spmm.hpp

```diff
@@ -60,11 +60,18 @@ void spmm(raft::resources const& handle,
 {
   bool is_row_major = detail::is_row_major(y, z);
 
+  // WARNING: The following copy is working around a bug in cusparse which causes an alignment issue
+  // and incorrect results. This bug is fixed in CUDA 12.5+ so this workaround shouldn't be removed
+  // until that version is supported.
+  auto z_tmp = raft::make_device_matrix<ValueType, IndexType>(handle, z.extent(0), z.extent(1));
```

Comment:
Suggestion from https://github.com/rapidsai/raft/pull/2185:

```suggestion
  // WARNING: The following copy is working around a bug in cusparse which causes an alignment issue
  // and incorrect results. This bug is fixed in CUDA 12.5+ so this workaround shouldn't be removed
  // until that version is supported.
  auto size = is_row_major ? (z.extent(0) - 1) * z.stride(0) + z.extent(1)
                           : (z.extent(1) - 1) * z.stride(1) + z.extent(0);
  rmm::device_uvector<ValueType> z_tmp(size, raft::resource::get_cuda_stream(handle));
```